src/box/memtx_tree.cc

```diff
@@ -709,6 +736,13 @@ tree_iterator_start(struct iterator *iterator, struct tuple **ret)
 		memtx_tree_iterator_prev(tree, &it->tree_iterator);
 		res = memtx_tree_iterator_get_elem(tree, &it->tree_iterator);
 	}
+	/* If we skip tuple, flag equals is not actual - need to refresh it. */
+	if (skip_eq && res != NULL && (type == ITER_EQ || type == ITER_REQ) &&
```

Comment:
Confusing variable name - `skip_eq`. Sounds like if it's set, the EQ check should be skipped, but actually it's vice versa. Better invert its meaning?

Reply:
Renamed to `skip_equal_tuple` and use it to change `GE` to `GT` and `LE` to `LT`